src/screens/DiscoveryScreenV2/fragment/DomainFragment.js

```diff
@@ -329,7 +329,7 @@ const DomainFragment = ({
         data={data}
         keyExtractor={(_, index) => index.toString()}
         renderItem={renderItemList}
-        onEndReached={() => fetchData()}
+        onEndReached={() => (firstData.length > 0 ? fetchData() : null)}
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The change made here is a good one. It adds a check to ensure that `fetchData()` is only called when there are elements in `firstData`. This can prevent unnecessary network requests when there's no more data to fetch, improving performance.
```diff
- onEndReached={() => fetchData()}
+ onEndReached={() => (firstData.length > 0 ? fetchData() : null)}
```
However, it would be better to use a more explicit check for array length. In JavaScript, `0` is falsy, so `firstData.length > 0` will return `false` if the length is `0`. But for readability and to avoid potential confusion, I suggest using `firstData.length !== 0` instead.
```diff
- onEndReached={() => (firstData.length > 0 ? fetchData() : null)}
+ onEndReached={() => (firstData.length !== 0 ? fetchData() : null)}
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->